backend/src/modules/idim-webservice/idim-webservice.service.ts

```diff
@@ -132,4 +142,109 @@ export class IdimWebserviceService {
             );
         }
     }
+
+    async verifyBceidUser(
+        userId: string,
+        requesterUserGuid: string,
+        requesterAccountTypeCode: string
+    ): Promise<HttpException | BCEIDUserResponse> {
+        if (!this.hasAllRequiredParams()) {
+            return new HttpException(
+                {
+                    error: 'Missing IDIM web service crednetials in the environment variables',
+                },
+                HttpStatus.INTERNAL_SERVER_ERROR
+            );
+        }
+
+        try {
+            const client = await this.getSoapClient();
+            // set xml schema parameters for the request
+            const requestData = {
+                accountDetailRequest: {
+                    onlineServiceId: this.idimWebServiceID,
+                    // who is sending the request
+                    requesterAccountTypeCode,
+                    requesterUserGuid,
+                    // who we search for, exact match userID
+                    userId,
+                    accountTypeCode: RequesterAccountTypeCode.Business,
+                },
+            };
+
+            return new Promise((resolve, reject) => {
+                client.BCeIDService.BCeIDServiceSoap.getAccountDetail(
+                    requestData,
+                    function (error, foundUser) {
+                        // this will be any error from the IDIM server side
+                        if (error) {
+                            return reject(
+                                new HttpException(
+                                    {
+                                        error:
+                                            'IDIM web service call error: ' +
+                                            error,
+                                    },
+                                    HttpStatus.INTERNAL_SERVER_ERROR
+                                )
+                            );
+                        }
+
+                        if (
+                            foundUser.getAccountDetailResult.code == 'Failed' &&
+                            foundUser.getAccountDetailResult.failureCode ==
+                                'NoResults'
```

Comment:
Is this soap method for BCEID (`getAccountDetailResult`) result response different than the IDIR (`searchInternalAccountResult`) search result response?
I meant, if the object return is **the same "type"** from the IDIM webservice, why do we handle them differently, since I see IDIR one we return a "reject" with search result's attribute but this one is different, it returns a normal response.

Reply:
Added the comment for the not found case below, also update the verifyIdirUser method, add comment there as well, thanks!!